oi/CodeGen.cpp

```diff
@@ -597,143 +597,167 @@ void addStandardTypeHandlers(std::string& code) {
-      }
+  size_t lastNonPaddingElement = getLastNonPaddingMemberIndex(c.members);
+  size_t pairs = 0;
 
-      if (thriftIssetMember != nullptr && thriftIssetMember != &member) {
-        // Return an additional VarInt before every field except for __isset
-        // itself.
-        pairs++;
-        if (i == lastNonPaddingElement) {
-          typeStaticType += "types::st::Pair<DB, types::st::VarInt<DB>, ";
-        } else {
-          typeStaticType += "types::st::VarInt<DB>, types::st::Pair<DB, ";
-        }
-      }
+  size_t emptySize = code.size();
+  for (size_t i = 0; i < lastNonPaddingElement + 1; i++) {
+    const auto& member = c.members[i];
+    if (member.name.starts_with(AddPadding::MemberPrefix)) {
+      continue;
+    }
 
-      typeStaticType +=
-          (boost::format("typename TypeHandler<DB, decltype(%1%::%2%)>::type") %
-           c.name() % member.name)
-              .str();
+    if (i != lastNonPaddingElement) {
+      code += "types::st::Pair<DB, ";
+      pairs++;
+    }
 
-      if (i != lastNonPaddingElement) {
-        typeStaticType += ", ";
+    if (thriftIssetMember != nullptr && thriftIssetMember != &member) {
+      // Return an additional VarInt before every field except for __isset
+      // itself.
+      pairs++;
+      if (i == lastNonPaddingElement) {
+        code += "types::st::Pair<DB, types::st::VarInt<DB>, ";
+      } else {
+        code += "types::st::VarInt<DB>, types::st::Pair<DB, ";
       }
     }
-    typeStaticType += std::string(pairs, '>');
 
-    if (typeStaticType.empty()) {
-      typeStaticType = "types::st::Unit<DB>";
+    code +=
+        (boost::format("typename TypeHandler<DB, decltype(%1%::%2%)>::type") %
+         c.name() % member.name)
+            .str();
+
+    if (i != lastNonPaddingElement) {
+      code += ", ";
     }
   }
+  code += std::string(pairs, '>');
 
-  // Generate the function body that walks the type. Uses the monadic
-  // `delegate()` form to handle each field except for the last. The last field
-  // is handled explicitly by passing it to `getSizeType`, as we must consume
-  // the entire type instead of delegating the next part.
-  std::string traverser;
-  {
-    for (size_t i = 0; i < lastNonPaddingElement + 1; i++) {
-      const auto& member = c.members[i];
-      if (member.name.starts_with(AddPadding::MemberPrefix)) {
-        continue;
-      }
+  if (code.size() == emptySize) {
+    code += "types::st::Unit<DB>";
+  }
+}
 
-      if (traverser.empty()) {
-        traverser = "auto ret = returnArg";
-      }
+void CodeGen::genClassTypeHandler(const Class& c, std::string& code) {
+  std::string helpers;
 
-      if (thriftIssetMember != nullptr && thriftIssetMember != &member) {
-        traverser += "\n  .write(getThriftIsset(t, " + std::to_string(i) + "))";
-      }
+  const Member* thriftIssetMember = nullptr;
+  if (const auto it = thriftIssetMembers_.find(&c);
+      it != thriftIssetMembers_.end()) {
+    thriftIssetMember = it->second;
```

Comment:
```suggestion
    const Member& thriftIssetMember = *it->second;
```
Minor, but a little simplification is possible now compared to the old code - no need to keep a pointer out of the scope of the if-block